From: Luca Weiss <luca.weiss@fairphone.com>
Date: Wed, 7 Dec 2022 09:36:52 +0100
Subject: [WIP] arm64: dts: qcom: sm7225-fairphone-fp4: Configure pmk8350
 thermals

One thermal sensor of the PMK8350 is used. Configure it.

Signed-off-by: Luca Weiss <luca.weiss@fairphone.com>

[WIP] qcom,adc-tm7 compatible not yet supported in mainline so bindings
subject to change and untested
---
 arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts | 25 +++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts b/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
index 18171c5..023b5de 100644
--- a/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
+++ b/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
@@ -107,6 +107,20 @@ active-config0 {
 				};
 			};
 		};
+
+		xo-therm-usr {
+			polling-delay-passive = <0>;
+			polling-delay = <0>;
+			thermal-sensors = <&pmk8350_adc_tm PMK8350_ADC7_AMUX_THM1_100K_PU>;
+
+			trips {
+				active-config0 {
+					temperature = <125000>;
+					hysteresis = <1000>;
+					type = "passive";
+				};
+			};
+		};
 	};
 };
 
@@ -511,6 +525,17 @@ conn-therm@1 {
 	};
 };
 
+&pmk8350_adc_tm {
+	status = "okay";
+
+	xo-therm@0 {
+		reg = <0>;
+		io-channels = <&pmk8350_vadc PMK8350_ADC7_AMUX_THM1_100K_PU>;
+		qcom,ratiometric;
+		qcom,hw-settle-time = <200>;
+	};
+};
+
 &pmk8350_rtc {
 	status = "okay";
 };
